### ISSUE TITLE
Question: Using Dynamic Providers, is there a built-in mechanism to store claims transformation/mapping per provider

### ISSUE THREAD

**Issue by omon77** [DuendeSoftware/Support#1381]
**Which version of Duende IdentityServer are you using?**
v7.0.6

**Which version of .NET are you using?**
.NET8

**Question**
Using Dynamic Providers feature (already configured), is there a built-in mechanism to store claims transformation/mapping configuration in [Properties](https://github.com/DuendeSoftware/IdentityServer/blob/e9860c6488f90e8fbc11a4452b9dd111dbfae933/src/Storage/Models/IdentityProvider.cs#L84) property of IdentityProvider type?
If it is not available or this is not the correct approach, what would be the correct approach?
We are an Enterprise license customer, and can reach out via email, if that's a better option.

**Comment by AndersAbel** [comment]
No, there is no built in support for driving claims transformation from per-provider configuration.

Our configuration system is however designed to facilitate such a scenario. Claims transformation can be done by the ClaimActions concept on the OpenIDConnectOptions. And we supply a way to configure the OpenIDConnectOptions from the provider object (including the properties).

See https://docs.duendesoftware.com/identityserver/v7/ui/login/dynamicproviders/#accessing-oidcprovider-data-in-iconfigurenamedoptions fore more details.

**Comment by omon77** [comment]
Thank you @AndersAbel !
